drivers/firmware/raspberrypi.c

```diff
@@ -198,8 +199,27 @@ static int rpi_firmware_notify_reboot(struct notifier_block *nb,
 	if (!fw)
 		return 0;
 
-	(void)rpi_firmware_property(fw, RPI_FIRMWARE_NOTIFY_REBOOT,
-				    0, 0);
+	// The partition id is the first parameter followed by zero or
+	// more flags indicating the reason for the reboot.
+	//
+	// 'tryboot': Sets a one-shot flag which is cleared upon reboot but
+	//            causes the config.txt [tryboot] conditional filter
```

Comment:
A lot of stuff has happened since this comment was written...

Reply:
Good point!